lib/ansible/module_utils/basic.py

```diff
@@ -1211,7 +1211,7 @@ def _check_required_arguments(self):
         missing = []
         for (k,v) in self.argument_spec.items():
             required = v.get('required', False)
-            if required and k not in self.params:
+            if required and self.params.get(k) is None:
```

Comment:
This one is probably going to cause a (extremely cornercase) bug.  A user could specify ```!!null``` in the playbook to set the parameter to None.  That should satisfy the required parameter (since the user set the param). We're only going to run into this if the module in question also thinks that None is a valid value for the parameter and specifies required=True.  Taken all together, I think this will be very unlikely to happen but I'm not sure we want to close out the possibility.